### ISSUE TITLE
Lighttpd config hates custom web pages.

### ISSUE THREAD

**Issue by Mcat12** [jacobsalmela/pi-hole#182]
I run a few other sites (tools and other personal stuff) from lighttpd (previously Apache). The current config redirects anything but the admin site to the ad blocking page, while at the same time using the ad blocking page as a 404 redirect. Since it redirects anything but the admin site, my pages get redirected. Does the config have to have both redirects, or can it just have the 404 redirect?

**Comment by dschaper** [comment]
I was working on a config that would only need the 404 redirect. I think that probably will work, as the html page names that are going to be sent for lookup are most likely not pages that you are serving up. The downside is that your 404's are going to be blank pages and not informative 404's for legitimate page not found traffic.

**Comment by dschaper** [comment]
Is there a way in 'lighttpd' to create a hierarchy of sites, like there is with 'nginx'? You can have named sites with domains configured with a stanza and then end with a 'default' site that is a catch all for non-virtual host sites. Kind of like an entire 404 site...

**Comment by Mcat12** [comment]
I just commented out the redirect if not admin, reloaded, and it seems to work.

**Comment by Mcat12** [comment]
Also, I added "Blocked by Pi-hole", as I was used to that from the dietpi version.

**Comment by dschaper** [comment]
I think we did something like that with another issue, https://github.com/jacobsalmela/pi-hole/issues/161#issuecomment-171179325

**Comment by Mcat12** [comment]
Do we need the special redirect, or can we just Add "Blocked by Pi-hole" to the blank page and only have the 404 redirect? I can make a pr if that's what we're going to do.

**Comment by dschaper** [comment]
From my thinking we only need a 404. Here's how I see the process:

1. Browser tries to fetch `http://ad.domain/ThisIsAnAd.gif`
2. Pi-Hole lookup returns local address for `ad.domain`
3. Browser requests `http://local/ThisIsAnAd.gif`
4. Local lighttpd looks for `ThisIsAnAd.gif` and doesn't find it, so it signals 404.
5. Local lighttpd configuration says 404's should serve `index.html` 
6. Browser renders 'http://ad.domain/ThisIsAnAd.gif` from source 'http://local/index.html`

**Comment by Mcat12** [comment]
Ok. I'll make a PR.

**Comment by jacobsalmela** [comment]
I think this should work well.  There isn't much of a speed difference between getting a 404 vs. 200.

**Comment by dschaper** [comment]
Resolved
